tests/manual/monkey/src/com/oracle/tools/fx/monkey/tools/StageTesterWindow.java

```diff
@@ -0,0 +1,286 @@
+            newStage.initModality(Modality.valueOf(modalityComboBox.getValue()));
+            newStage.initDefaultHeaderButtons(defaultHeaderButtonsCheckBox.isSelected());
+            newStage.setTitle(titleTextField.getText());
+            newStage.setAlwaysOnTop(alwaysOnTopCheckBox.isSelected());
+            newStage.setResizable(resizableCheckBox.isSelected());
+            newStage.setIconified(iconifiedCheckBox.isSelected());
+            newStage.setMaximized(maximizedCheckBox.isSelected());
+            newStage.setFullScreen(fullScreenCheckBox.isSelected());
+            newStage.setFullScreenExitHint(fullScreenExitHintTextField.getText().isEmpty()
+                                           ? null : fullScreenExitHintTextField.getText());
+
+            if (newStage.getModality() != Modality.NONE) {
+                newStage.initOwner(StageTesterWindow.this);
+            }
+
+            Parent root = switch (headerBarComboBox.getValue().toLowerCase(Locale.ROOT)) {
+                case "simple" -> createSimpleHeaderBarRoot(newStage, !defaultHeaderButtonsCheckBox.isSelected());
+                case "split" -> createSplitHeaderBarRoot(newStage, !defaultHeaderButtonsCheckBox.isSelected());
+                default -> new BorderPane(createWindowActions(newStage));
+            };
+
+            var scene = new Scene(root);
+            scene.setNodeOrientation(NodeOrientation.valueOf(nodeOrientationComboBox.getValue()));
+
+            newStage.setWidth(800);
+            newStage.setHeight(500);
+            newStage.setScene(scene);
+            newStage.show();
+        });
+
+        var root = new BorderPane(pane);
+        root.setPadding(new Insets(20));
+        root.setBottom(showStageButton);
+        BorderPane.setAlignment(showStageButton, Pos.CENTER);
+        BorderPane.setMargin(showStageButton, new Insets(30, 0, 0, 0));
+
+        initModality(Modality.APPLICATION_MODAL);
+        initOwner(owner);
+        setScene(new Scene(root));
+        setTitle("Stage Tester");
+    }
+
+    private Parent createSimpleHeaderBarRoot(Stage stage, boolean customWindowButtons) {
+        var headerBar = new HeaderBar();
+        headerBar.setBackground(Background.fill(Color.LIGHTSKYBLUE));
+        headerBar.setCenter(new TextField() {{ setPromptText("Search..."); }});
+
+        var sizeComboBox = new ComboBox<>(FXCollections.observableArrayList("Small", "Medium", "Large"));
+        sizeComboBox.getSelectionModel().select(0);
+
+        Runnable updateMinHeight = () -> headerBar.setMinHeight(
+            switch (sizeComboBox.getValue().toLowerCase(Locale.ROOT)) {
+                case "large" -> 80;
+                case "medium" -> 40;
+                default -> headerBar.getMinSystemHeight();
+            });
+
+        sizeComboBox.valueProperty().subscribe(event -> updateMinHeight.run());
+        headerBar.minSystemHeightProperty().subscribe(event -> updateMinHeight.run());
+        headerBar.setLeading(new Button("✨"));
```

Comment:
minor: this is nice, but can we replace this with non-emoji character please?  (messes up the line spacing in Eclipse)